src/Commands/core/TwigUnusedCommand.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drush\Commands\core;
+
+use Consolidation\OutputFormatters\FormatterManager;
+use Consolidation\OutputFormatters\Options\FormatterOptions;
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+use Drupal\Core\PhpStorage\PhpStorageFactory;
+use Drupal\Core\Template\TwigEnvironment;
+use Drush\Attributes as CLI;
+use Drush\Boot\BootstrapManager;
+use Drush\Commands\AutowireTrait;
+use Drush\Formatters\FormatterTrait;
+use Drush\Utils\StringUtils;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Path;
+use Symfony\Component\Finder\Finder;
+
+#[AsCommand(
+    name: self::NAME,
+    description: 'Find potentially unused Twig templates.',
+    aliases: ['twu']
+)]
+#[CLI\FilterDefaultField(field: 'template')]
+final class TwigUnusedCommand extends Command
+{
+    use AutowireTrait;
+    use FormatterTrait;
+
+    public const NAME = 'twig:unused';
+
+    public function __construct(
+        protected readonly FormatterManager $formatterManager,
+        protected readonly BootstrapManager $bootstrapManager,
+        protected readonly TwigEnvironment $twig,
+        private readonly LoggerInterface $logger
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->addArgument('searchpaths', InputArgument::REQUIRED, 'A comma delimited list of paths to recursively search')
+            // Usages can't have a description with plain Console :(. Use setHelp() if desired as per  https://github.com/symfony/symfony/issues/45050
+            ->addUsage('twig:unused /var/www/mass.local/docroot/modules/custom,/var/www/mass.local/docroot/themes/custom')
+            ->setHelp('Immediately before running this command, web crawl your entire web site. Or use your Production PHPStorage dir for comparison.');
+        $this->addFormatterOptions(RowsOfFields::class);
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): int
```

Comment:
Why does this command have its own copy of execute?

Reply:
I removed execute() from FormatterTrait and into the Command. See https://github.com/drush-ops/drush/pull/6135/commits/6216ad68e4f0a41013e879ecb206877abaf57be5. This design has a bit less magic and a bit more boilerplate for the command. I think its a decent tradeoff.

Reply:
If format commands have to copy boilerplate code, it seems that there's little need to factor out `doExecute`. I think I still favor putting execute in the trait, with `doExecute` renamed to something like `getOutputData()`.

I often prefer to Write Everything Twice rather than make all code perfectly DRY, but in this case I think it's useful to have the boilerplate code in the trait.

How about this:

Every command's `execute()` method calls a differently-named method in the formatter trait. Something like this:
```
    public function execute(InputInterface $input, OutputInterface $output): int
    {
         $rowsOfFields = $this->iCanNameThisMethodWhateverILike();
         $this->writeFormattedOutput($input, $output, $rowsOfFields);
         return self::SUCCESS;
    }
```
This eliminates hiding the implementation of `execute()` in the trait, while still providing the boilerplate code for commands to use.

I am still thinking that formatterOptions should be set up in `configure()`, for consistency with Symfony Console patterns (and also to give us a way to add the `--format` flag automatically), but `setInput()` would still be called by `writeFormattedOutput()`.

Reply:
[That proposal is very close to where the code is now.](https://github.com/drush-ops/drush/pull/6135/files#diff-adb922e5dacf604fec2453710d6cd73c99d8ea00a026bf04718863037f6213adR59-R65) FormatterTrait no longer care about the doExecute() method name or its return type hint. 

In the current code getFormatteroptions() is called during configure() and again during execute(). I think you are proposing that it get called once during configure() and stored in a property in FormatterTrait? That works for me.

I'm not sure in your proposal where filtering is happenning. In the link above, its in the call to `$this->alterResult($data, $input)`

Reply:
I made a commit which implements what I think you re suggesting - https://github.com/drush-ops/drush/pull/6135/commits/f923a801dd50d4aa17fd2390790336bc962785a3

Reply:
Yeah, just a minor adjustment, but I like it a lot better.